client/src/main/java/org/schlunzis/kurtama/client/server/JarServer.java

```diff
@@ -0,0 +1,177 @@
+    private static final String JAR_URL_VERSION_REPLACEMENT = "${kurtama-version}";
+    private static final String JAR_PATH = "kurtama-server.jar";
+
+    private final VersionManager versionManager;
+
+    private final Semaphore mutex = new Semaphore(1);
+    private String javaExecutable = null;
+
+    JarServer(VersionManager versionManager, LogSink logSink) {
+        super(logSink);
+        this.versionManager = versionManager;
+    }
+
+    @Override
+    public boolean testRequirements() {
+        if (!testUsingJavaHome()) {
+            log.info("Testing using JAVA_HOME failed, trying PATH");
+            return testUsingPath();
+        }
+        return true;
+    }
+
+    private boolean testUsingJavaHome() {
+        log.info("Testing for java installation using JAVA_HOME");
+        String javaHome = System.getenv("JAVA_HOME");
+        if (javaHome == null) {
+            log.info("JAVA_HOME not set");
+            return false;
+        }
+
+        String testExecutable = javaHome + File.separator + "bin" + File.separator + "java";
+        if (test(testExecutable)) {
+            javaExecutable = testExecutable;
+            return true;
+        }
+        return false;
+    }
+
+    private boolean testUsingPath() {
+        String path = System.getenv("PATH");
+        if (path == null) {
+            log.info("PATH not set");
+            return false;
+        }
+
+        String[] pathEntries = path.split(File.pathSeparator);
+        for (String pathEntry : pathEntries) {
+            String testExecutable = pathEntry + File.separator + "java";
+            if (test(testExecutable)) {
+                javaExecutable = testExecutable;
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private boolean test(String javaExecutable) {
+        log.info("Testing for java installation using executable: {}", javaExecutable);
+        ProcessBuilder processBuilder = new ProcessBuilder(javaExecutable, "-version");
+        try {
+            Process process = processBuilder.start();
+            StreamGobbler streamGobbler = new StreamGobbler(process.getErrorStream());
+            executor.submit(streamGobbler);
+
+            int exitCode = process.waitFor();
+            String[] output = streamGobbler.getOutput();
+            Arrays.stream(output).forEach(log::info);
+            int javaVersion = extractJavaVersion(output);
+            log.info("Java version: {}", javaVersion);
+
+            return exitCode == 0 && javaVersion >= MINIMUM_JAVA_VERSION;
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            log.error("Error while testing JAR requirements", e);
+            return false;
+        } catch (IOException e) {
+            log.error("Error while testing JAR requirements", e);
```

Comment:
```suggestion
            log.debug("Error while testing JAR requirements", e);
```

This exception is likely a file not found `IOException`. But this is expected for most of the locations we are searching for the java executable. So I think, we should only log the exceptions as debug messages and not as real errors. They don't really need to show up in the logs unless we want to make sure, the search is working. Only after all locations have been checked and not executable has been found, we can log an error.